src/figury/AnimPanel.java

```diff
@@ -63,7 +105,14 @@ void animate() {
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		device.drawImage(image, 0, 0, null);
-		buffer.clearRect(0, 0, getWidth(), getHeight());
+		SwingUtilities.invokeLater(new Runnable() {
```

Comment:
to nie jest konieczne - `actionPerformed` jest odpalone we właściwym wątku, a opóźnienie nie jest potrzebne